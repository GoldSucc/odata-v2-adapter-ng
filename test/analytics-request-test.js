"use strict";

const fs = require("fs");
const supertest = require("supertest");

const env = require("./_env");
const util = require("./_env/util");
const init = require("./_env/data/init");

let context;
let request;

describe("analytics-request", () => {
  beforeAll(async () => {
    context = await env("analyticsmodel", 0, init);
    request = supertest(context.app);
  });

  afterAll(() => {
    env.end(context);
  });

  it("GET request with grouping and aggregation", async () => {
    let response = await util.callRead(request, "/v2/analytics/Header?$select=currency,stock");
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["currency","stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "EUR",
            stock: 25,
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["currency","stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "USD",
            stock: 17,
          },
        ],
      },
    });
    response = await util.callRead(
      request,
      `/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["currency","stock"]}')`
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        __metadata: {
          uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["currency","stock"]}')`,
          type: "test.AnalyticsService.Header",
        },
        currency: "USD",
        stock: 17,
      },
    });
  });

  it("GET request with grouping and aggregating many", async () => {
    const response = await util.callRead(request, "/v2/analytics/Header?$select=country,currency,stock,price");
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"country":"'Germany'","currency":"'EUR'"},"value":["country","currency","stock","price"]}')`,
              type: "test.AnalyticsService.Header",
            },
            country: "Germany",
            currency: "EUR",
            stock: 10,
            price: "12.12",
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"country":"'Spain'","currency":"'EUR'"},"value":["country","currency","stock","price"]}')`,
              type: "test.AnalyticsService.Header",
            },
            country: "Spain",
            currency: "EUR",
            stock: 15,
            price: "11.11",
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"country":"'Texas'","currency":"'USD'"},"value":["country","currency","stock","price"]}')`,
              type: "test.AnalyticsService.Header",
            },
            country: "Texas",
            currency: "USD",
            stock: 17,
            price: "10.165000000000001",
          },
        ],
      },
    });
  });

  it("GET request with grouping and aggregation and order by", async () => {
    const response = await util.callRead(
      request,
      "/v2/analytics/Header?$select=currency,stock&$top=4&$orderby=stock asc"
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["currency","stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "USD",
            stock: 17,
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["currency","stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "EUR",
            stock: 25,
          },
        ],
      },
    });
  });

  it("GET request with grouping and aggregation and filter", async () => {
    let response = await util.callRead(
      request,
      "/v2/analytics/Header?$select=stock,currency&$top=4&$filter=currency eq 'USD'"
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["stock","currency"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "USD",
            stock: 17,
          },
        ],
      },
    });
    response = await util.callRead(
      request,
      "/v2/analytics/Header?$select=stock,currency&$top=4&$filter=(currency eq 'USD' or currency eq 'EUR')"
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["stock","currency"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "EUR",
            stock: 25,
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["stock","currency"]}')`,
              type: "test.AnalyticsService.Header",
            },
            currency: "USD",
            stock: 17,
          },
        ],
      },
    });
    response = await util.callRead(
      request,
      '/v2/analytics/Header(aggregation\'{"key":{"currency":"\'EUR\'"},"value":["stock","currency"]}\')'
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        __metadata: {
          uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["stock","currency"]}')`,
          type: "test.AnalyticsService.Header",
        },
        currency: "EUR",
        stock: 25,
      },
    });

    response = await util.callRead(request, "/v2/analytics/Header?$select=stock&$top=4&$filter=currency eq 'USD'");
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            stock: 17,
          },
        ],
      },
    });
    response = await util.callRead(
      request,
      "/v2/analytics/Header?$select=stock&$top=4&$filter=(currency eq 'USD' or currency eq 'EUR')"
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        results: [
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            stock: 25,
          },
          {
            __metadata: {
              uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'USD'"},"value":["stock"]}')`,
              type: "test.AnalyticsService.Header",
            },
            stock: 17,
          },
        ],
      },
    });
    response = await util.callRead(
      request,
      '/v2/analytics/Header(aggregation\'{"key":{"currency":"\'EUR\'"},"value":["stock"]}\')'
    );
    expect(response.body).toBeDefined();
    expect(response.body).toEqual({
      d: {
        __metadata: {
          uri: `http://${response.request.host}/v2/analytics/Header(aggregation'{"key":{"currency":"'EUR'"},"value":["stock"]}')`,
          type: "test.AnalyticsService.Header",
        },
        stock: 25,
      },
    });
  });
});
