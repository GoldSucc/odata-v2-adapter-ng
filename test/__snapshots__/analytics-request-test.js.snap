// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics-request GET $metadata 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<edmx:Edmx Version=\\"1.0\\" xmlns:edmx=\\"http://schemas.microsoft.com/ado/2007/06/edmx\\" xmlns:m=\\"http://schemas.microsoft.com/ado/2007/08/dataservices/metadata\\" xmlns:sap=\\"http://www.sap.com/Protocols/SAPData\\">
  <edmx:Reference Uri=\\"https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Aggregation.V1.xml\\" xmlns:edmx=\\"http://docs.oasis-open.org/odata/ns/edmx\\">
    <edmx:Include Alias=\\"Aggregation\\" Namespace=\\"Org.OData.Aggregation.V1\\"/>
  </edmx:Reference>
  <edmx:Reference Uri=\\"https://sap.github.io/odata-vocabularies/vocabularies/Analytics.xml\\" xmlns:edmx=\\"http://docs.oasis-open.org/odata/ns/edmx\\">
    <edmx:Include Alias=\\"Analytics\\" Namespace=\\"com.sap.vocabularies.Analytics.v1\\"/>
  </edmx:Reference>
  <edmx:Reference Uri=\\"https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Capabilities.V1.xml\\" xmlns:edmx=\\"http://docs.oasis-open.org/odata/ns/edmx\\">
    <edmx:Include Alias=\\"Capabilities\\" Namespace=\\"Org.OData.Capabilities.V1\\"/>
  </edmx:Reference>
  <edmx:Reference Uri=\\"https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml\\" xmlns:edmx=\\"http://docs.oasis-open.org/odata/ns/edmx\\">
    <edmx:Include Alias=\\"Core\\" Namespace=\\"Org.OData.Core.V1\\"/>
  </edmx:Reference>
  <edmx:Reference Uri=\\"https://sap.github.io/odata-vocabularies/vocabularies/UI.xml\\" xmlns:edmx=\\"http://docs.oasis-open.org/odata/ns/edmx\\">
    <edmx:Include Alias=\\"UI\\" Namespace=\\"com.sap.vocabularies.UI.v1\\"/>
  </edmx:Reference>
  <edmx:DataServices m:DataServiceVersion=\\"2.0\\">
    <Schema Namespace=\\"test.AnalyticsService\\" xmlns=\\"http://schemas.microsoft.com/ado/2008/09/edm\\">
      <EntityContainer Name=\\"EntityContainer\\" m:IsDefaultEntityContainer=\\"true\\">
        <EntitySet Name=\\"Header\\" EntityType=\\"test.AnalyticsService.Header\\" sap:semantics=\\"aggregate\\"/>
        <EntitySet Name=\\"HeaderDisabledAnalytics\\" EntityType=\\"test.AnalyticsService.HeaderDisabledAnalytics\\" sap:semantics=\\"aggregate\\"/>
        <EntitySet Name=\\"HeaderItem\\" EntityType=\\"test.AnalyticsService.HeaderItem\\"/>
        <EntitySet Name=\\"HeaderLine\\" EntityType=\\"test.AnalyticsService.HeaderLine\\"/>
        <AssociationSet Name=\\"HeaderDisabledAnalytics_Items\\" Association=\\"test.AnalyticsService.HeaderDisabledAnalytics_Items\\">
          <End Role=\\"HeaderDisabledAnalytics\\" EntitySet=\\"HeaderDisabledAnalytics\\"/>
          <End Role=\\"HeaderItem\\" EntitySet=\\"HeaderItem\\"/>
        </AssociationSet>
        <AssociationSet Name=\\"HeaderItem_header\\" Association=\\"test.AnalyticsService.HeaderItem_header\\">
          <End Role=\\"HeaderItem\\" EntitySet=\\"HeaderItem\\"/>
          <End Role=\\"Header\\" EntitySet=\\"Header\\"/>
        </AssociationSet>
        <AssociationSet Name=\\"HeaderItem_NextItem\\" Association=\\"test.AnalyticsService.HeaderItem_NextItem\\">
          <End Role=\\"HeaderItem\\" EntitySet=\\"HeaderItem\\"/>
          <End Role=\\"HeaderItem1\\" EntitySet=\\"HeaderItem\\"/>
        </AssociationSet>
        <AssociationSet Name=\\"HeaderLine_item\\" Association=\\"test.AnalyticsService.HeaderLine_item\\">
          <End Role=\\"HeaderLine\\" EntitySet=\\"HeaderLine\\"/>
          <End Role=\\"HeaderItem\\" EntitySet=\\"HeaderItem\\"/>
        </AssociationSet>
      </EntityContainer>
      <EntityType Name=\\"Header\\" sap:semantics=\\"aggregate\\">
        <Key>
          <PropertyRef Name=\\"ID__\\"/>
        </Key>
        <Property Name=\\"ID__\\" Type=\\"Edm.String\\" Nullable=\\"false\\" sap:sortable=\\"false\\" sap:filterable=\\"false\\"/>
        <Property Name=\\"ID\\" Type=\\"Edm.Guid\\"/>
        <Property Name=\\"description\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"country\\" Type=\\"Edm.String\\" sap:aggregation-role=\\"dimension\\"/>
        <Property Name=\\"currency\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"stock\\" Type=\\"Edm.Int32\\" sap:aggregation-role=\\"measure\\"/>
        <Property Name=\\"price\\" Type=\\"Edm.Decimal\\" Scale=\\"2\\" Precision=\\"12\\" sap:aggregation-role=\\"measure\\"/>
        <NavigationProperty Name=\\"Items\\" Relationship=\\"test.AnalyticsService.HeaderItem_header\\" FromRole=\\"Header\\" ToRole=\\"HeaderItem\\"/>
      </EntityType>
      <EntityType Name=\\"HeaderDisabledAnalytics\\" sap:semantics=\\"aggregate\\">
        <Key>
          <PropertyRef Name=\\"ID__\\"/>
        </Key>
        <Property Name=\\"ID__\\" Type=\\"Edm.String\\" Nullable=\\"false\\" sap:sortable=\\"false\\" sap:filterable=\\"false\\"/>
        <Property Name=\\"ID\\" Type=\\"Edm.Guid\\"/>
        <Property Name=\\"description\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"country\\" Type=\\"Edm.String\\" sap:aggregation-role=\\"dimension\\"/>
        <Property Name=\\"currency\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"stock\\" Type=\\"Edm.Int32\\" sap:aggregation-role=\\"measure\\"/>
        <Property Name=\\"price\\" Type=\\"Edm.Decimal\\" Scale=\\"2\\" Precision=\\"12\\" sap:aggregation-role=\\"measure\\"/>
        <NavigationProperty Name=\\"Items\\" Relationship=\\"test.AnalyticsService.HeaderDisabledAnalytics_Items\\" FromRole=\\"HeaderDisabledAnalytics\\" ToRole=\\"HeaderItem\\"/>
      </EntityType>
      <EntityType Name=\\"HeaderItem\\">
        <Key>
          <PropertyRef Name=\\"ID\\"/>
        </Key>
        <Property Name=\\"ID\\" Type=\\"Edm.Guid\\" Nullable=\\"false\\"/>
        <Property Name=\\"name\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"description\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"startAt\\" Type=\\"Edm.DateTimeOffset\\" Precision=\\"7\\"/>
        <Property Name=\\"endAt\\" Type=\\"Edm.DateTimeOffset\\" Precision=\\"7\\"/>
        <NavigationProperty Name=\\"header\\" Relationship=\\"test.AnalyticsService.HeaderItem_header\\" FromRole=\\"HeaderItem\\" ToRole=\\"Header\\"/>
        <Property Name=\\"header_ID\\" Type=\\"Edm.Guid\\"/>
        <NavigationProperty Name=\\"NextItem\\" Relationship=\\"test.AnalyticsService.HeaderItem_NextItem\\" FromRole=\\"HeaderItem\\" ToRole=\\"HeaderItem1\\"/>
        <Property Name=\\"NextItem_ID\\" Type=\\"Edm.Guid\\"/>
        <Property Name=\\"assoc_header_ID\\" Type=\\"Edm.Guid\\"/>
        <NavigationProperty Name=\\"Lines\\" Relationship=\\"test.AnalyticsService.HeaderLine_item\\" FromRole=\\"HeaderItem\\" ToRole=\\"HeaderLine\\"/>
      </EntityType>
      <EntityType Name=\\"HeaderLine\\">
        <Key>
          <PropertyRef Name=\\"ID\\"/>
        </Key>
        <Property Name=\\"ID\\" Type=\\"Edm.Guid\\" Nullable=\\"false\\"/>
        <Property Name=\\"name\\" Type=\\"Edm.String\\"/>
        <Property Name=\\"value\\" Type=\\"Edm.Double\\"/>
        <NavigationProperty Name=\\"item\\" Relationship=\\"test.AnalyticsService.HeaderLine_item\\" FromRole=\\"HeaderLine\\" ToRole=\\"HeaderItem\\"/>
        <Property Name=\\"item_ID\\" Type=\\"Edm.Guid\\"/>
      </EntityType>
      <Association Name=\\"HeaderDisabledAnalytics_Items\\">
        <End Role=\\"HeaderDisabledAnalytics\\" Type=\\"test.AnalyticsService.HeaderDisabledAnalytics\\" Multiplicity=\\"1\\">
          <OnDelete Action=\\"Cascade\\"/>
        </End>
        <End Role=\\"HeaderItem\\" Type=\\"test.AnalyticsService.HeaderItem\\" Multiplicity=\\"*\\"/>
      </Association>
      <Association Name=\\"HeaderItem_header\\">
        <End Role=\\"HeaderItem\\" Type=\\"test.AnalyticsService.HeaderItem\\" Multiplicity=\\"*\\"/>
        <End Role=\\"Header\\" Type=\\"test.AnalyticsService.Header\\" Multiplicity=\\"0..1\\">
          <OnDelete Action=\\"Cascade\\"/>
        </End>
      </Association>
      <Association Name=\\"HeaderItem_NextItem\\">
        <End Role=\\"HeaderItem\\" Type=\\"test.AnalyticsService.HeaderItem\\" Multiplicity=\\"*\\"/>
        <End Role=\\"HeaderItem1\\" Type=\\"test.AnalyticsService.HeaderItem\\" Multiplicity=\\"0..1\\"/>
        <ReferentialConstraint>
          <Principal Role=\\"HeaderItem1\\">
            <PropertyRef Name=\\"ID\\"/>
          </Principal>
          <Dependent Role=\\"HeaderItem\\">
            <PropertyRef Name=\\"NextItem_ID\\"/>
          </Dependent>
        </ReferentialConstraint>
      </Association>
      <Association Name=\\"HeaderLine_item\\">
        <End Role=\\"HeaderLine\\" Type=\\"test.AnalyticsService.HeaderLine\\" Multiplicity=\\"*\\"/>
        <End Role=\\"HeaderItem\\" Type=\\"test.AnalyticsService.HeaderItem\\" Multiplicity=\\"0..1\\">
          <OnDelete Action=\\"Cascade\\"/>
        </End>
        <ReferentialConstraint>
          <Principal Role=\\"HeaderItem\\">
            <PropertyRef Name=\\"ID\\"/>
          </Principal>
          <Dependent Role=\\"HeaderLine\\">
            <PropertyRef Name=\\"item_ID\\"/>
          </Dependent>
        </ReferentialConstraint>
      </Association>
      <Annotations Target=\\"test.AnalyticsService.Header\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Aggregation.ApplySupported\\">
          <Record Type=\\"Aggregation.ApplySupportedType\\">
            <PropertyValue Property=\\"PropertyRestrictions\\" Bool=\\"true\\"/>
          </Record>
        </Annotation>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.EntityContainer/Header\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Aggregation.ApplySupported\\">
          <Record Type=\\"Aggregation.ApplySupportedType\\">
            <PropertyValue Property=\\"PropertyRestrictions\\" Bool=\\"true\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.DeleteRestrictions\\">
          <Record Type=\\"Capabilities.DeleteRestrictionsType\\">
            <PropertyValue Property=\\"Deletable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.InsertRestrictions\\">
          <Record Type=\\"Capabilities.InsertRestrictionsType\\">
            <PropertyValue Property=\\"Insertable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.UpdateRestrictions\\">
          <Record Type=\\"Capabilities.UpdateRestrictionsType\\">
            <PropertyValue Property=\\"Updatable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.Header/ID__\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"UI.Hidden\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.Header/country\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Dimension\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.Header/stock\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Measure\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.Header/price\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Measure\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.HeaderDisabledAnalytics\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Aggregation.ApplySupported\\">
          <Record Type=\\"Aggregation.ApplySupportedType\\">
            <PropertyValue Property=\\"PropertyRestrictions\\" Bool=\\"true\\"/>
          </Record>
        </Annotation>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.EntityContainer/HeaderDisabledAnalytics\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Aggregation.ApplySupported\\">
          <Record Type=\\"Aggregation.ApplySupportedType\\">
            <PropertyValue Property=\\"PropertyRestrictions\\" Bool=\\"true\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.DeleteRestrictions\\">
          <Record Type=\\"Capabilities.DeleteRestrictionsType\\">
            <PropertyValue Property=\\"Deletable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.InsertRestrictions\\">
          <Record Type=\\"Capabilities.InsertRestrictionsType\\">
            <PropertyValue Property=\\"Insertable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
        <Annotation Term=\\"Capabilities.UpdateRestrictions\\">
          <Record Type=\\"Capabilities.UpdateRestrictionsType\\">
            <PropertyValue Property=\\"Updatable\\" Bool=\\"false\\"/>
          </Record>
        </Annotation>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.HeaderDisabledAnalytics/ID__\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"UI.Hidden\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.HeaderDisabledAnalytics/country\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Dimension\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.HeaderDisabledAnalytics/stock\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Measure\\" Bool=\\"true\\"/>
      </Annotations>
      <Annotations Target=\\"test.AnalyticsService.HeaderDisabledAnalytics/price\\" xmlns=\\"http://docs.oasis-open.org/odata/ns/edm\\">
        <Annotation Term=\\"Analytics.Measure\\" Bool=\\"true\\"/>
      </Annotations>
    </Schema>
  </edmx:DataServices>
</edmx:Edmx>"
`;
